code/game/objects/items/weapons/storage/fancy.dm

```diff
@@ -425,6 +431,44 @@
 	item_state = "Fpacket"
 	cigarette_to_spawn = /obj/item/clothing/mask/smokable/cigarette/oracle
 
+/obj/item/storage/box/fancy/cigarettes/case
+	name = "cigarette case"
+	desc = "A simple cigarette case for the aspiring chain-smoker."
+	icon_state = "cigarettecase"
+	item_state = "cigarettecase"
+	storage_type = "case"
+	drop_sound = 'sound/items/drop/weldingtool.ogg'
+	pickup_sound = 'sound/items/pickup/weldingtool.ogg'
+	use_sound = 'sound/items/storage/briefcase.ogg'
+	storage_slots = 14
+	max_storage_space = 14 //A janky but functional solution since I could not figure out how to increase the maximum storage to match how much the case is designed to hold in any other way
+	cigarette_to_spawn = null
```

Comment:
You need to fix line 315. This is causing init to spawn new null(src) which is breaking your checks.

Just put an if(cigarette_to_spawn) check there.